specs/request-handler-plugins.md

```diff
@@ -0,0 +1,116 @@
+# Support a "plug-in" model for request handlers
+
+To allow developers to provide unique handling behaviors for different APIs the Microsoft Graph will have a plug-in model to dynamically load request handlers. This will provide separation of concerns between the software components intercepting network request and those providing responses.
+To enable developers building handlers an abstractions library will be distributed on NuGet.
+
+## History
+
+| Version | Date | Comments | Author |
+| ------- | -------- | ----- | --- |
+| 1.0 | 2022-12-19 | Initial specifications | @gavinbarron |
+
+## Configuration file
+
+The existing `appsettings.json` file will be used to provide configuration. Both for determining which request handlers are loaded into the proxy and providing configuration for each of the loaded request handlers. The set of request handlers to be loaded will be specified as an array of objects at the root of the configuration file in the `handlers` property. Each handler must supply a `configSection`, `handerPath`, `name`, and `urlsToWatch` in the handlers array.
+- `configSection` is a string for which there must be a matching object property at the root of the configuration file. If a handler does not provide any configuration options then an empty object must be supplied.
```

Comment:
Why do we require specifying an empty object if the plugin doesn't need any configuration? Isn't it making it harder than necessary for the users to bring in plugins?

Reply:
Totally valid, it makes implementation a little more complex in that we need to deal with a null config section, but that's simple enough.